src/server/luaentity_sao.cpp

```diff
@@ -402,6 +392,20 @@ std::string LuaEntitySAO::getDescription()
 void LuaEntitySAO::setHP(s32 hp, const PlayerHPChangeReason &reason)
 {
 	m_hp = rangelim(hp, 0, U16_MAX);
+
+	sendPunchCommand();
```

Comment:
This might be incorrect. HP changes are not always punches - especially when the HP increases.

Reply:
https://github.com/minetest/minetest/blob/ef22c0206f225dbccd67bff9fb888867c63039b3/src/client/content_cao.cpp#L1849-L1885
Looks like the client code handles this correctly.

Reply:
i agree with @SmallJoker don't trust the client

Reply:
@nerzhul This is not about trust.
---

I see. The function simply got an unfortunate naming. It's just about sending the HP update packet; nothing else. A better name for `AO_CMD_PUNCHED` would be `AO_CMD_SET_HP`.